### ISSUE TITLE
Using Style function to override styling

### ISSUE THREAD

**Issue by bustamantedev** [Shopify/restyle#58]
### Description

I have created a custom Card Component that looks like this. I'm using `Pressable` as the base component.
```typescript
export type CardProps = BoxProps<Theme> &
  VariantProps<Theme, 'cardVariants'> &
  PressableProps
export const Card = createRestyleComponent<CardProps, Theme>(
  [createVariant({ themeKey: 'cardVariants' })],
  createBox<Theme>(Pressable)
)
```

This has allowed me to use the Card like this:

```typescript
<Card
      variant="horizontal"
      m="xl"
      onPress={() => {
        console.log('pressed card')
      }}>
```

It works but I had the question if it is possible to take advantage of the Style function provided by `Pressable` [See here](https://reactnative.dev/docs/pressable#example) to override styles with it:

This works:

```typescript
<Card
      variant="horizontal"
      m="xxl"
      onPress={() => {
        console.log('card pressed')
      }}
      style={{ backgroundColor: 'red' }}>
```

This doesn't:

```typescript
<Card
      variant="horizontal"
      m="xxl"
      onPress={() => {
        console.log('card pressed')
      }}
      style={({pressed}) => ({ backgroundColor: 'red' })}>
```

I'm not sure if there's a workaround for this, but it would be cool if possible.

Thanks in advance

**Comment by VictorPulzz** [comment]
Same issue, restyle overwrite style props, and I can't use this: 
```
style={({ pressed }) => [
          {
            backgroundColor: pressed
              ? 'rgb(210, 230, 255)'
              : 'white'
          },
          styles.wrapperCustom
        ]}
```

**Comment by CodyBrouwers** [comment]
I have a small fix for this issue in this PR

- https://github.com/Shopify/restyle/pull/138

Might be more complicated then it first appears but let's see what the maintainers think. Works great for me so far though!

**Comment by sbalay** [comment]
Fixed as part of v2.1.0

**Comment by bustamantedev** [comment]
> Fixed as part of v2.1.0

Thanks a lot for this, guys! Awesome work

**Comment by a-eid** [comment]
you do lose the ability to hook into your theme inside the style function right ?
